File: src/Website/Models/Isaac/GameplayEvent.cs
using System.Collections.Generic;
using Website.Models.Isaac.Enums;

namespace Website.Models.Isaac
{
    public class GameplayEvent
    {
        public int Id { get; set; }
        public GameplayEventType EventType { get; set; }
        public List<IsaacResource> Resources { get; set; }
    }
}

File: src/Website/Ts/Models/gameplay-event.ts
import { GameplayEventType } from "../Enums/gameplay-event-type";
import { IsaacResource } from "./isaac-resource";

export interface GameplayEvent {
    id: number;
    event_type: GameplayEventType;
    r1: IsaacResource;
    r2: IsaacResource | null;
    r3: number | null;
    action: number;
    in_consequence_of: IsaacResource | null;
    run_number: number;
    player: number | null;
    floor_number: number;
    submission: number;
}

